app/uk/gov/hmrc/customs/declaration/services/StatusResponseValidationService.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.customs.declaration.services
+
+import javax.inject.{Inject, Singleton}
+import org.joda.time.format.ISODateTimeFormat
+import org.joda.time.{DateTime, DateTimeZone}
+import uk.gov.hmrc.customs.declaration.logging.DeclarationsLogger
+import uk.gov.hmrc.customs.declaration.model.BadgeIdentifier
+
+import scala.xml.NodeSeq
+
+@Singleton
+class StatusResponseValidationService @Inject() (declarationsLogger: DeclarationsLogger, declarationsConfigService: DeclarationsConfigService) {
+
+  def validate(xml: NodeSeq, badgeIdentifier: BadgeIdentifier): Boolean = {
+    val declarationNode = xml \ "responseDetail" \ "declarationManagementInformationResponse" \ "declaration"
+    validateBadgeIdentifier(declarationNode, badgeIdentifier) && validateReceivedDate(declarationNode)
+  }
+
+  private def validateBadgeIdentifier(declarationNode: NodeSeq, badgeIdentifier: BadgeIdentifier): Boolean = {
+    extractField(declarationNode, "communicationAddress").fold(false)({ communicationAddress =>
+      validateCommunicationAddress(communicationAddress.head) && badgeIdentifier.value == extractBadgeIdentifier(communicationAddress.head)
+    })
+  }
+
+  private def extractBadgeIdentifier(communicationAddress: String) = {
+    communicationAddress.split(":").last
+  }
+
+  private def validateCommunicationAddress(communicationAddress: String): Boolean = {
+    val regexString = "hmrcgwid:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{16}:[0-9a-zA-Z]{10}"
```

Comment:
`val regexString = ...` could be extracted as a top level field as it is effectively a constant

Reply:
I'll look at this as it was my code

Reply:
I think the regex is not needed when talking to the team. We just need to check it matches the badgeId we get sent. Will remove as it add more code that is not needed